src/main/groovy/life/qbic/utils/NanoporeParser.groovy

```diff
@@ -30,29 +30,32 @@ class NanoporeParser {
         try {
             validateJsonForSchema(json, JSON_SCHEMA)
             //Step3: return valid json as Map
-            parseMetaData(convertedDirectory)
+            parseMetaData(convertedDirectory, directory)
             return convertedDirectory
         } catch (ValidationException validationException) {
             log.error("Specified directory could not be validated")
-            log.error(validationException.getMessage())
-            log.debug(validationException)
+            // we have to fetch all validation exceptions
+            def causes = validationException.getCausingExceptions().collect{ it.message  }.join("\n")
+            log.error(causes)
             throw validationException
         }
     }
 
 
-    private static Map parseMetaData(Map convertedDirectory) {
+    private static Map parseMetaData(Map convertedDirectory, Path root) {
         convertedDirectory.get("children").each { measurement ->
             def reportFile = measurement["children"].find {it["name"].contains("report") && it["file_type"] == "md"}
             def summaryFile = measurement["children"].find {it["name"].contains("final_summary") && it["file_type"] == "txt"}
-            def metadata = readMetaData(reportFile as Map, summaryFile as Map)
+            def metadata = readMetaData(reportFile as Map, summaryFile as Map, root)
             measurement["metadata"] = metadata
         }
         return convertedDirectory
     }
 
-    private static Map readMetaData(Map<String, String> reportFile, Map<String, String> summaryFile) {
-        def report = new File(reportFile["path"].toString()).readLines().iterator()
+    private static Map readMetaData(Map<String, String> reportFile, Map<String, String> summaryFile, Path root) {
+        def report = new File(root.toString() + reportFile["path"].toString())
```

Comment:
Dangerous Path joining.
```suggestion
        def report = new File(Paths.get(root.toString(),reportFile["path"].toString()))
```

Reply:
Maybe you can elaborate why it is dangerous? So we can learn ;)

Reply:
See comment below ;)